flink-tests/src/test/java/org/apache/flink/test/example/client/JobRetrievalITCase.java

```diff
@@ -119,6 +121,11 @@ public void run() {
 		lock.release();
 
 		resumingThread.join();
+
+		Throwable exception = error.get();
+		if (exception != null) {
+			throw new AssertionError(exception);
+		}
```

Comment:
Is this any different from using `Assert.assertNull(error.get())`? Not that it makes any difference but just to know why you picked throwing the exception here. Personally I think that the assertNull seems more explicit when you read the code and it is less verbose.